### PR TITLE
Add logging

### PR DESCRIPTION
**Pull request by bessman**
This PR adds a logging system. It uses printf and therefore outputs on SYSCALLS_UART_BUS.

The low-level part of the logging system does not output directly; instead it stores log entries in a ring buffer which the application must periodically poll.

## Summary by Sourcery

Introduce a two-layer logging framework with a buffer-based platform layer and a printf-backed system layer; integrate logging into the main loop, update build scripts, add comprehensive unit tests, and supply an example.

New Features:
- Add low-level platform logging API (LOG_LL_* functions and macros) using a ring buffer and service flag
- Add high-level system logging macros (LOG_ERROR/WARN/INFO/DEBUG) over stdio/UART syscalls
- Implement LOG_service_platform to forward buffered platform log entries to the system logger
- Integrate logging into main.c by replacing raw printf calls and polling platform logs

Build:
- Update CMakeLists to include platform logging sources in the platform library and to register logging unit tests

Documentation:
- Add example file demonstrating platform and system logging usage

Tests:
- Add comprehensive Unity test suite for platform logging buffer operations, levels, service behavior, and edge cases

### CODE REVIEW

File: src/platform/logging_ll.c [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        circular_buffer_read(
+            &g_LOG_LL_buffer, (uint8_t *)&entry->length, sizeof(entry->length)
+        ) != sizeof(entry->length)) {
+        return false;
+    }
+
+    /* Validate length */
+    if (entry->length >= sizeof(entry->message)) {
+        entry->length = sizeof(entry->message) - 1;
+    }
```
[the remark this answers is not in the document]
Comment (bessman) [reply src/platform/logging_ll.c:134]: We can't log from within the logging machinery itself.

File: tests/test_logging.c [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+void test_LOG_LL_convenience_macros(void)
+{
+    // Test the convenience macros defined in the header
+    LOG_LL_ERROR("Error macro test");
+    LOG_LL_WARN("Warning macro test");
+    LOG_LL_INFO("Info macro test");
+    LOG_LL_DEBUG("Debug macro test");
+
+    // Verify the macros work correctly
+    LOG_LL_Entry entry;
```
[the remark this answers is not in the document]
Comment (bessman) [reply tests/test_logging.c:364]: There is such a test already: test_LOG_LL_write_with_format_string

File: tests/test_logging.c [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+    // Act
+    LOG_LL_write(LOG_LL_INFO, "%s", long_message);
+
+    // Assert - Message should be truncated to fit buffer
+    LOG_LL_Entry entry;
+    TEST_ASSERT_TRUE(LOG_LL_read_entry(&entry));
+    TEST_ASSERT_EQUAL(LOG_LL_INFO, entry.level);
+    TEST_ASSERT_EQUAL(LOG_LL_MAX_MESSAGE_SIZE - 1, entry.length);
+    TEST_ASSERT_EQUAL('\0', entry.message[entry.length]);
+}
```
[the remark this answers is not in the document]
Comment (bessman) [reply tests/test_logging.c]: The size of a log entry is an implementation detail that should not be tested. Testing exact buffer fill is out of scope for the logging module.
